client/src/client.rs

```diff
@@ -344,6 +344,8 @@ impl_query_output! {
 )]
 #[display(fmt = "{}@{torii_url}", "key_pair.public_key()")]
 pub struct Client {
+    /// Unique id of the blockchain. Used for simple replay attack protection.
+    pub chain_id: u16,
```

Comment:
Is using a `u16` here a good idea?

It would hard for the deployers to select a unique value, as there isn't really a defined order for various iroha chains unlike Etherium which can just use 1 for mainnet, and other networks just don't matter as much.

Even if deployers would select perfectly random numbers (which they most probably won't), due to birthday paradox they would collide quite often. If my calculations are right, with 300 randomly chosen `u16` numbers, with around 50% at least two will collide.

I propose to make the chain id a string, which is more natural for unordered structures.

If dealing with variable-sized data structure here is a problem, we can hash the user-provided config value into 64 bits with blake2 and then use that.

Reply:
sure, it was a stupid choice. Even though `chain_id` is a publicly available information it should be best defined as `UUID` because there is no way to coordinate uniqueness of `chain_id` across different chains

Reply:
Yeah, a UUID would also work here

Reply:
changed to `String` now